cmd/soroban-rpc/internal/config/config.go

```diff
@@ -52,22 +52,29 @@ func (cfg *Config) Init(cmd *cobra.Command) error {
 	return cfg.flags().Init(cmd)
 }
 
-// Values applied as defaults -> flags/env -> config file
+// We start with the defaults
 func (cfg *Config) SetValues() error {
 	if err := cfg.loadDefaults(); err != nil {
 		return err
 	}
 
-	cfg.Bind()
+	// Then we load from the cli flags and environment variables
 	if err := cfg.loadFlags(); err != nil {
 		return err
 	}
 
+	// If we specified a config file, we load that
 	if cfg.ConfigPath != "" {
 		// Merge in the config file flags
 		if err := cfg.loadConfigPath(); err != nil {
 			return err
 		}
+
+		// Load from cli flags and environment variables again, to overwrite what we
+		// got from the config file
+		if err := cfg.loadFlags(); err != nil {
+			return err
+		}
```

Comment:
Would it be equivalent for us to `loadFlags` once after the `if cfg.ConfigPath != ""` block instead of twice, once before and once after during?

Reply:
We could. We specifically need to load the configPath first, in order to load the config file.